File: backend/app/constants/application_status.py
"""
Ready2Go CRM — Application Status Constants

All valid lifecycle stages for an applicant's visa application.
"""

INQUIRY = "inquiry"
DOCUMENTS_PENDING = "documents_pending"
DOCUMENTS_SUBMITTED = "documents_submitted"
APPLICATION_PROCESSING = "application_processing"
UNDER_REVIEW = "under_review"
INTERVIEW_SCHEDULED = "interview_scheduled"
VISA_APPROVED = "visa_approved"
VISA_REJECTED = "visa_rejected"
APPROVED = "approved"
VISA_ISSUED = "visa_issued"
REJECTED = "rejected"
CANCELLED = "cancelled"
COMPLETED = "completed"

# For validation checks
ALL_STATUSES = (
    INQUIRY,
    DOCUMENTS_PENDING,
    DOCUMENTS_SUBMITTED,
    APPLICATION_PROCESSING,
    UNDER_REVIEW,
    INTERVIEW_SCHEDULED,
    VISA_APPROVED,
    VISA_REJECTED,
    APPROVED,
    VISA_ISSUED,
    REJECTED,
    CANCELLED,
    COMPLETED,
)

File: frontend/src/constants/applicationStatus.js
/**
 * Ready2Go CRM — Application Status Constants
 *
 * Single source of truth for all status values and their visual config.
 * Used across StatusBadge, FilterPanel, ApplicantForm, and tables.
 */

export const STATUSES = [
  'inquiry',
  'documents_pending',
  'documents_submitted',
  'application_processing',
  'under_review',
  'interview_scheduled',
  'visa_approved',
  'visa_rejected',
  'approved',
  'visa_issued',
  'rejected',
  'cancelled',
  'completed',
];

export const STATUS_LABELS = {
  inquiry: 'Inquiry',
  documents_pending: 'Docs Pending',
  documents_submitted: 'Docs Submitted',
  application_processing: 'Processing',
  under_review: 'Under Review',
  interview_scheduled: 'Interview Set',
  visa_approved: 'Visa Approved',
  visa_rejected: 'Visa Rejected',
  approved: 'Approved',
  visa_issued: 'Visa Issued',
  rejected: 'Rejected',
  cancelled: 'Cancelled',
  completed: 'Completed',
};

export const STATUS_OPTIONS = STATUSES.map((s) => ({
  value: s,
  label: STATUS_LABELS[s],
}));

export default STATUS_OPTIONS;
